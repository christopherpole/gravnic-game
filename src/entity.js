const NONE = 0;
const FLOOR = 1;
const BLOCK = 2;

module.exports = {
  NONE,
  FLOOR,
  BLOCK,
};
