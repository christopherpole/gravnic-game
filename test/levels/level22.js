import { MOVE_DOWN, MOVE_RIGHT, ENTITIES, levelIsComplete, makeMoves } from '../../src/index';

describe('Level 22', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 2 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#0000FF', id: 1 },
        },
        {
          staticEntity: {
            entityId: ENTITIES.BARRIER.id,
            color: '#008000',
            powered: false,
            id: 3,
          },
          movableEntity: null,
        },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 5 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#0000FF', id: 4 },
        },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 7 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#ff0000', id: 6 },
        },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 9 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#800080', id: 8 },
        },
        {
          staticEntity: { entityId: ENTITIES.BARRIER.id, powered: false, id: 10 },
          movableEntity: null,
        },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 12 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#800080', id: 11 },
        },
      ],
      [
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.BARRIER.id, powered: true, id: 13 },
          movableEntity: null,
        },
        {
          staticEntity: {
            entityId: ENTITIES.BARRIER.id,
            color: '#ff0000',
            powered: true,
            id: 14,
          },
          movableEntity: null,
        },
        {
          staticEntity: {
            entityId: ENTITIES.BARRIER.id,
            color: '#8B4513',
            powered: true,
            id: 15,
          },
          movableEntity: null,
        },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
      [
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 16 }, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 18 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#ff0000', id: 17 },
        },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 19 }, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_DOWN, MOVE_RIGHT]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
