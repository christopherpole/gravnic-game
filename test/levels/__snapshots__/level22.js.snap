// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 22 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 11,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 6,
          },
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 11,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "fading": true,
            "id": 6,
          },
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "fading": true,
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 11,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 11,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 11,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 11,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "fading": true,
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "fading": true,
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "fading": true,
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "fading": true,
            "id": 11,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#008000",
            "entityId": "BARRIER",
            "id": 3,
            "powered": false,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 10,
            "powered": false,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BARRIER",
            "id": 13,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#ff0000",
            "entityId": "BARRIER",
            "id": 14,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "color": "#8B4513",
            "entityId": "BARRIER",
            "id": 15,
            "powered": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 19,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
  ],
]
`;
